Fix docs regarding deduplication periods [KVL-1194] (#11738)
digital-asset/daml: docs/source/app-dev/app-arch.rst

The ledger will then guarantee that commands with the same :ref:`change ID <change-id>` will generate a rejection within the effective deduplication period.

To use command deduplication, you should:

- Use generous values for the deduplication duration. It should be large enough such that you can assume the command was permanently lost if the effective deduplication period has passed and you still don’t observe any effect of the command on the ledger (i.e. you don't see a transaction with the command ID via the :ref:`transaction service <transaction-service>`).
- Make sure you set command IDs deterministically, that is to say: the "same" command must use the same command ID. This is useful for the recovery procedure after an application crash/restart, in which the application inspects the state of the ledger (e.g. via the :ref:`Active contracts service <active-contract-service>`) and sends commands to the ledger. When using deterministic command IDs, any commands that had been sent before the application restart will be discarded by the ledger to avoid duplicate submissions.

> - If you are not sure whether a command was submitted successfully, just resubmit it with the same deduplication duration. If the new command was submitted within the effective deduplication period, the duplicate submission will safely be ignored. If the effective deduplication period has passed, you can assume the command was lost or rejected and a new submission is justified.

Reviewer: Should we correct the `will safely be ignored` here as well?

Author: Good point, corrected for consistency.

.. note:: Effective deduplication period

>   In case of ledgers with command deduplication across participants, the effective deduplication period may be longer than the one specified in the command.

Reviewer 2: The effective deduplication period may always be longer than the one specified in the command. I don't think this is specific to VMBC ledgers.
```suggestion
  The effective deduplication period may be longer than the one specified in the command, but it never is shorter.
```
